pkg/kubectl/cmd/util/factory.go

```diff
@@ -251,8 +251,11 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			return generator, ok
 		},
 		CanBeExposed: func(kind string) error {
-			if kind != "ReplicationController" && kind != "Service" && kind != "Pod" {
-				return fmt.Errorf("invalid resource provided: %v, only a replication controller, service or pod is accepted", kind)
+			switch kind {
+			case "ReplicationController", "Service", "Pod":
+				// nothing to do here
+			default:
+				return fmt.Errorf("cannot expose a %s", kind)
```

Comment:
I kind of like how previously the error noted what can be accepted. Could that be added back into the error message?